### PR TITLE
Add a completion provider for 11labs text2Speech

### PR DESCRIPTION
**Pull request by coffeeorgreentea**
## What Changed:

A plugin for eleven labs has been added which adds a completion provider for text2Speech.

## How to test:

Setup your .env.local to include the plugin 'elevenlabs'
Use the new text to speech node under the Audio category.

## Additional information:

It is currently returning a blob array, and until we have a storage direction, will not play well with connectors such as discord.

### CODE REVIEW

**Comment by fire** [comment]
What do you think about squashing so there's one commit when ready for merge?

File: plugins/elevenlabs/client/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  {
+    type: InputControl,
+    dataKey: 'voice_id',
+    name: 'Voice ID',
+    icon: 'moon',
+    defaultValue: '',
+  },
+];
+
+// Object containing all input controls for different completion types
```
Comment (fire) [inline plugins/elevenlabs/client/src/index.ts:37]: Last time mentioning sentence case (preferring) versus other.

File: plugins/elevenlabs/server/src/functions/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// DOCUMENTED 
+/** 
+ * Exporting necessary modules for external use.
+ */
+export * from './textToSpeech';
```
Comment (fire) [inline plugins/elevenlabs/server/src/functions/index.ts:5]: No end of the sentence line.

**Comment by coffeeorgreentea** [comment]
> What do you think about squashing so there's one commit when ready for merge?

I prefer to keep it the way it is for my own organization and for when I improve it.

File: plugins/elevenlabs/server/src/functions/textToSpeech.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+// UNDOCUMENTED
```
Comment (fire) [inline plugins/elevenlabs/server/src/functions/textToSpeech.ts:1]: How do we get this documented?

**Review by fire — COMMENTED** [review]
@coffeeorgreentea ~~What is the test plan for this? How do I use this?~~ reading the test plan

File: plugins/elevenlabs/client/src/index.ts (hunk 1 above)
Comment (coffeeorgreentea) [reply plugins/elevenlabs/client/src/index.ts:37]: im just matching the existing plugin formats

File: packages/core/shared/src/types.ts [hunk 4]
```diff
@@ -226,6 +226,7 @@ export type PubSubEvents = {
   $SAVE_SPELL_DIFF: (tabId: string) => string
   $CREATE_MESSAGE_REACTION_EDITOR: (tabId: string) => string
   $CREATE_PLAYTEST: (tabId: string) => string
+  $CREATE_MEDIAWINDOW: (tabId: string) => string
```
[the remark this answers is not in the document]
Comment (coffeeorgreentea) [reply packages/core/shared/src/types.ts:229]: this is actually part of a sidequest i was doing and has no use. got commited by mistake

File: packages/core/shared/src/nodes/audio/textToSpeech.ts [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+            node.inputs.delete(socket.socket)
+          }
+        })
+        inputSockets.forEach(socket => {
+          node.addInput(new Rete.Input(socket.socket, socket.name, socket.type))
+        })
+        lastInputSockets = inputSockets
+      }
+      // update output sockets
```
[the remark this answers is not in the document]
Comment (coffeeorgreentea) [reply packages/core/shared/src/nodes/audio/textToSpeech.ts:118]: https://github.com/Oneirocom/Magick/pull/724#discussion_r1176982376

File: packages/core/shared/src/nodes/audio/textToSpeech.ts [hunk 6]
```diff
@@ -0,0 +1,196 @@
+// UNDOCUMENTED
```
[the remark this answers is not in the document]
Comment (coffeeorgreentea) [reply packages/core/shared/src/nodes/audio/textToSpeech.ts:1]: Undocumented means its not in the documentation yet

[the remark this answers is not in the document]
Comment (fire) [reply packages/core/shared/src/nodes/audio/textToSpeech.ts:1]: Will circle back.

File: packages/core/shared/src/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (fire) [reply packages/core/shared/src/types.ts:229]: Thanks, resolve this by removing it I guess?

File: packages/core/shared/src/nodes/audio/textToSpeech.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fire) [reply packages/core/shared/src/nodes/audio/textToSpeech.ts:118]: Will circle back later.

File: plugins/elevenlabs/client/src/index.ts (hunk 1 above)
Comment (fire) [reply plugins/elevenlabs/client/src/index.ts:37]: Will circle back later.

**Pull request merged**